processor/memorylimiterprocessor/memorylimiter.go

```diff
@@ -226,8 +234,16 @@ func (ml *memoryLimiter) startMonitoring() {
 
 	ml.refCounter++
 	if ml.refCounter == 1 {
+		ml.waitGroup.Add(1)
 		go func() {
-			for range ml.ticker.C {
+			defer ml.waitGroup.Done()
+
+			for {
+				select {
+				case <-ml.ticker.C:
+				case <-ml.closed:
```

Comment:
Is it possible to get into some weird case (since the ticker is closed before `ml.closed`) where we get an empty value from `m.ticker.C` and run `ml.checkMemLimits()` when we're trying to shutdown? Should we do an `ok` check?

Reply:
> since the ticker is closed before ml.closed

It's not closed. Stopping the ticker doesn't close its channel